app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -389,7 +389,7 @@ def find_host(h, ems_id)
         # host = Host.find_by_hostname(hostname) has a risk of creating duplicate hosts
         # allow a deleted EMS to be re-added an pick up old orphaned hosts
         _log.debug("EMS ID: #{ems_id} Host database lookup - hostname: [#{h[:hostname]}] IP: [#{h[:ipaddress]}] ems_ref: [#{h[:ems_ref]}]")
-        found = look_up_host(h[:hostname], h[:ipaddress], :ems_ref => h[:ems_ref])
+        found = look_up_host(h[:hostname], h[:ipaddress], :ems_id => ems_id)
```

Comment:
Skipping hosts that didn't have the same ems_ref is invalid since importing old hosts into a new vCenter will always give them new ems_refs.  We already check above for `ems_id + ems_ref` regardless of the hostname so I believe any ems_ref checks in look_up_host aren't correct and should be removed in the future.